src/mlpack/methods/neighbor_search/ns_model.hpp

```diff
@@ -37,6 +49,127 @@ struct NSModelName<FurthestNeighborSort>
   static const std::string Name() { return "furthest_neighbor_search_model"; }
 };
 
+class MonoSearchVisitor : public boost::static_visitor<void>
```

Comment:
@rcurtin Sure!! I am here for this! I agree it can be confusing for someone with no knowledge of boost variants. I will add documentation.
Thanks!
Marcos